packages/salesforcedx-vscode-visualforce-next/typings/vscode.proposed.d.ts

```diff
@@ -0,0 +1,316 @@
+    mtime: number;
+    // atime: number;
+    size: number;
+    type: FileType;
+  }
+
+  // todo@joh discover files etc
+  export interface FileSystemProvider {
+    onDidChange?: Event<FileChange[]>;
+
+    root: Uri;
+
+    // more...
+    //
+    utimes(resource: Uri, mtime: number, atime: number): Thenable<FileStat>;
+
+    stat(resource: Uri): Thenable<FileStat>;
+
+    read(
+      resource: Uri,
+      offset: number,
+      length: number,
+      progress: Progress<Uint8Array>
+    ): Thenable<number>;
+
+    // todo@remote
+    // offset - byte offset to start
+    // count - number of bytes to write
+    // Thenable<number> - number of bytes actually written
+    write(resource: Uri, content: Uint8Array): Thenable<void>;
+
+    // todo@remote
+    // Thenable<FileStat>
+    move(resource: Uri, target: Uri): Thenable<FileStat>;
+
+    // todo@remote
+    // helps with performance bigly
+    // copy?(from: Uri, to: Uri): Thenable<void>;
+
+    // todo@remote
+    // Thenable<FileStat>
+    mkdir(resource: Uri): Thenable<FileStat>;
+
+    readdir(resource: Uri): Thenable<[Uri, FileStat][]>;
+
+    // todo@remote
+    // ? merge both
+    // ? recursive del
+    rmdir(resource: Uri): Thenable<void>;
+    unlink(resource: Uri): Thenable<void>;
+
+    // todo@remote
+    // create(resource: Uri): Thenable<FileStat>;
+  }
+
+  export namespace workspace {
+    export function registerFileSystemProvider(
+      authority: string,
+      provider: FileSystemProvider
+    ): Disposable;
+  }
+
+  export namespace window {
+    export function sampleFunction(): Thenable<any>;
+  }
+
+  /**
+     * The contiguous set of modified lines in a diff.
+     */
+  export interface LineChange {
+    readonly originalStartLineNumber: number;
+    readonly originalEndLineNumber: number;
+    readonly modifiedStartLineNumber: number;
+    readonly modifiedEndLineNumber: number;
+  }
+
+  export namespace commands {
+    /**
+       * Registers a diff information command that can be invoked via a keyboard shortcut,
+       * a menu item, an action, or directly.
+       *
+       * Diff information commands are different from ordinary [commands](#commands.registerCommand) as
+       * they only execute when there is an active diff editor when the command is called, and the diff
+       * information has been computed. Also, the command handler of an editor command has access to
+       * the diff information.
+       *
+       * @param command A unique identifier for the command.
+       * @param callback A command handler function with access to the [diff information](#LineChange).
+       * @param thisArg The `this` context used when invoking the handler function.
+       * @return Disposable which unregisters this command on disposal.
+       */
+    export function registerDiffInformationCommand(
+      command: string,
+      callback: (diff: LineChange[], ...args: any[]) => any,
+      thisArg?: any
+    ): Disposable;
+  }
+
+  /**
```

Comment:
This line and below are the proposed color API that we are using. The above are not being used.

This vscode.proposed.d.ts was copied from this particular version https://github.com/Microsoft/vscode/commit/a7ad71be3bf9c1db4ad247d9a1ac39738540c13a#diff-4878e8b25d41f2ab7c2715852b1b97aa